src/scripts/main.js

```diff
@@ -1 +1,73 @@
 'use strict';
+
+const firstPromise = new Promise((resolve, reject) => {
+  const handler = () => {
+    resolve();
+    document.removeEventListener('click', resolve);
```

Comment:
You are removing the event listener with 'resolve' as the handler, but you actually added the event listener with 'handler' as the handler. This could lead to potential memory leaks as the event listener will not be removed properly.